.gitlab-ci.yml

```diff
@@ -218,6 +207,90 @@ build_windows_msi_x64:
     paths:
     - .omnibus/pkg
 
+
+
+#
+# image_build
+#
+
+
+# build the agent6 image
+build_agent6:
+  stage: image_build
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/devtools/dind/build:latest
+  tags: [ "runner:dind", "size:large" ]
+  script: [ "# noop" ]
+  variables:
+    DD_DIND_BUILD_CONTEXT: Dockerfiles/agent6
+    DD_DIND_IMAGE: 486234852809.dkr.ecr.us-east-1.amazonaws.com/datadog-agent/agent:$CI_COMMIT_SHA
+    DD_DIND_ARTEFACTS: "true"
+
+# build the dogstatsd image
+build_dogstatsd:
+  stage: image_build
+  dependencies:
+    - build_dogstatsd_static-deb_x64
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/devtools/dind/build:latest
+  tags: [ "runner:dind", "size:large" ]
+  script: [ "# noop" ]
+  variables:
+    DD_DIND_BUILD_CONTEXT: Dockerfiles/dogstatsd/alpine
+    DD_DIND_IMAGE: 486234852809.dkr.ecr.us-east-1.amazonaws.com/datadog-agent/dogstatsd:$CI_COMMIT_SHA
+    DD_DIND_ARTEFACTS: "true"
+
+#
+# image_deploy
+#
+agent6_docker_hub:
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/devtools/dind/tag:latest
+  tags: [ "runner:dind", "size:large" ]
+  stage: image_deploy
+  when: manual
+  only:
+    - master
+    - tags
+  variables:
+    DD_DIND_TAG_SOURCE: 486234852809.dkr.ecr.us-east-1.amazonaws.com/datadog-agent/agent:$CI_COMMIT_SHA
+    DD_DIND_DEST_REGISTRY_TYPE: DOCKER_HUB
+    DD_DIND_TAG_DEST: datadog/agent:$CI_COMMIT_TAG
+  script: [ "# noop" ]
+
+agent6_dev_docker_hub:
```

Comment:
I'd love to see `master` automatically pushed to `agent-dev` & `dogstatsd-dev`, can we configure that (auto on master, manual on other branches), or does that require duplicating the step?

Reply:
Yes we can add this but we'll have to duplicate the step, Gitlab doesn't handle multiple `when/only` triggers for a single step

Reply:
Added 👍